### PR TITLE
Update documentation for s3, switch non secret values to go environmental variables.

### PR DESCRIPTION
**Pull request by pguinard-public-com**

### CODE REVIEW

File: cmd/terraform-registry/main.go [hunk 1]
```diff
@@ -115,9 +118,6 @@ func main() {
 	// Load environment variables here!
 	gitHubToken = os.Getenv("GITHUB_TOKEN")
 
-	awsRegion = os.Getenv("AWS_REGION")
```
Comment (pguinard-public-com) [inline cmd/terraform-registry/main.go:118]: Breaking change!

File: README.md [hunk 2]
```diff
@@ -31,6 +31,7 @@ Supported Terraform protocols:
 Supported backends:
 - `MemoryStore`: a dumb in-memory store currently only used for testing
 - [`GitHubStore`](#github-store): queries the GitHub API for modules, version tags and SSH download URLs
+- [`S3Store`](#s3-store): queries a s3 bucket for modules, version tags and HTTPS download URLs
```
Comment (nilspublic) [inline README.md:34]: It should be `queries an S3 bucket`...

File: README.md [hunk 3]
```diff
@@ -111,6 +112,28 @@ Command line arguments:
 - `-github-owner-filter`: GitHub org/user repository filter
 - `-github-topic-filter`: GitHub topic repository filter
 
+### S3 Store
+
+This store uses AWS s3 as a backend. A query for the module address
```
Comment (nilspublic) [inline README.md:117]: `S3` is the product name

File: README.md [hunk 4]
```diff
@@ -111,6 +112,28 @@ Command line arguments:
...
+`namespace/name/provider` will be translated directly to a s3 bucket key.
```
Comment (nilspublic) [inline README.md:118]: `an S3`

File: README.md [hunk 5]
```diff
@@ -111,6 +112,28 @@ Command line arguments:
...
+Requests for module sources are redirected to s3, authentication is handled by s3.
```
Comment (nilspublic) [inline README.md:119]: I think this should be elaborated a bit, something like:
`access permissions are handled by AWS policies` or even more verbose. Meant to make the next user consider why access is failing and/or set it up.

Comment (nilspublic) [inline README.md:119]: I think this should be elaborated a bit, something like:
`access permissions are handled by AWS policies` or even more verbose. Meant to make the next user consider why access is failing and/or set it up.

File: README.md (hunk 2 above)
Comment (pguinard-public-com) [reply README.md:34]: I'll capitalize s3 but it's `a` not `an`.

**Review by nilspublic — APPROVED** [review]

**Pull request merged**